### ISSUE TITLE
For listed versions, do not allow submissions with a version number lower than the current public version

### ISSUE THREAD

**Issue by wagnerand** [mozilla/addons-server#19788]
### Describe the problem and steps to reproduce it:

On the listed channel, we are in control of the update path and Firefox only ever updates to versions considered newer than the one you have installed. Users would never update to it, new users would likely update to an older version (that has a higher version number), reviewers review code that never reaches users and developers are confused why their latest submission doesn't get usage. Therefore, we should fail on a list submission that uses a lower version number than the current public listed version. 

### What happened?

E.g. when your public version is 1.0, you can submit 0.9.

### What did you expect to happen?

E.g. when your public version is 1.0, you cannot submit 0.9.

### Anything else we should know?
This should happen for channels only where we have a defined and controlled update path, which is listed at this point.

**Comment by willdurand** [comment]
Note that this will require version normalization, e.g. `1.0 == 1.0.0 == 1.0.0.0`

**Comment by willdurand** [comment]
> Note that this will require version normalization, e.g. `1.0 == 1.0.0 == 1.0.0.0` but AMO currently treats them as 3 different versions.

To be clear, Andreas asked to prevent submission of a version lower than the current public version and I'd like to expand that to lower **or equal** to the current public version.

**Comment by ioanarusiczki** [comment]
@willdurand @eviljeff 

I'm still looking around , so far I've noticed for the lang packs errors which don't seem right: 

![error2](https://user-images.githubusercontent.com/33448286/199459780-3efd91ba-dbd2-40cc-85a9-5b3adcdc762e.png)
![langpackapierror](https://user-images.githubusercontent.com/33448286/199459790-81070ce4-f0ff-49c6-a85a-93a0e05bc28c.png)
![langpackserror](https://user-images.githubusercontent.com/33448286/199459793-f872ca59-06a7-4a37-a6a1-0a98dd27e698.png)
![Screenshot 2022-11-02 113910](https://user-images.githubusercontent.com/33448286/199460031-649bed43-167a-4a08-9ec5-223f39244bd9.png)

If I change the ff version from 107 to 108 -> the new version can be uploaded

**Comment by willdurand** [comment]
@ioanarusiczki this is caused by https://github.com/mozilla/addons-server/issues/19788 as explained by @eviljeff on Matrix. I suggest creating a new issue because that will need to be resolved.

**Comment by ioanarusiczki** [comment]
@willdurand I filed https://github.com/mozilla/addons-server/issues/19849

**Comment by eviljeff** [comment]
#19852 reverts #19820 - there's no easy fix here and landing a complicated change is risky as a cherry-pick, given the potential impact on releng and other add-on developers.

**Comment by ioanarusiczki** [comment]
@eviljeff 
I'm not getting the error right now on AMO dev. 

I tried with an [extension](https://addons-dev.allizom.org/en-US/developers/addon/test1_versions/versions), I submitted listed version 4.0.5.6 then 1.0 then 0.1 and again 1.0.0.0
Also with a[ lang pack](https://addons-dev.allizom.org/en-US/developers/addon/basque-language-pack_test3/versions)

**Comment by eviljeff** [comment]
> @eviljeff I'm not getting the error right now on AMO dev.
> 
> I tried with an [extension](https://addons-dev.allizom.org/en-US/developers/addon/test1_versions/versions), I submitted listed version 4.0.5.6 then 1.0 then 0.1 and again 1.0.0.0 Also with a[ lang pack](https://addons-dev.allizom.org/en-US/developers/addon/basque-language-pack_test3/versions)

The restriction only applies to the current public version - it looks like those versions weren't approved so wouldn't have been considered.  (Actually, seems like that add-on doesn't have any public versions)

> P.S> I see a tag failed to deploy on dev, could be the reason why

https://addons-dev.allizom.org/__version__ is showing https://github.com/mozilla/addons-server/commit/e84c7b578d95f5290a8bd38dc4b5ebd54b5b488e to me

**Comment by ioanarusiczki** [comment]
@eviljeff 

I approved 3.0.0.6 and then submitted 3.0.0.5 https://addons-dev.allizom.org/en-US/developers/addon/higher-version/versions
Same with a lang pack https://addons-dev.allizom.org/en-US/developers/addon/basque-language-pack_approved/versions

**Comment by ioanarusiczki** [comment]
@eviljeff 

I did some testing on AMO dev and the signing API, it looks good, I am going to mark it verified only after https://github.com/mozilla/addons-server/issues/19964 lands when I'll re-check some scenarios.
